### DISCUSSION TITLE
Riven constantly downloads the same media

### DISCUSSION THREAD

**Discussion by profucius** [I-am-PUID-0/DMB#91] [Q&A]
Hey there, I'm opening this discussion as an extension/replacement to my other discussion at #69 (nice)

TL;DR Riven is re-downloading media constantly, as if it never knows it exists in Plex.

---

I thought originally that that issue was related to Riven not triggering my Plex to update after downloading a media to RD/Zurg, but now I think that may be unrelated. Even after triggering a manual scan in Plex, it seems Riven is re-downloading media for media that already exists in Plex. I cannot tell why this is happening, so I am not sure if I should discuss it in the Riven discord, or if its something wrong with my setup, so I thought I'd ask here.

I noticed when looking at my Debrid Media Manager (same dev as Zurg) that I have a ton of duplicate hash/title entries. Like, over a thousand added per day. I resolve them by deleting the duplicates, but then a day later it's happened again. I looked at the Riven logs in my DMB stack in portainer, and I see Riven backend subprocess scanning Torrentio for tv shows that are already in Plex. Would it be helpful if I shared a log here?

Again the thought I had in #69 was that this might be happening because I have my Plex installed on Windows but DMB installed on a separate machine in Proxmox. But now I am thinking that since Riven has access to the Plex Server DB via the URL, I don't see why this would be the issue; Riven should be seeing that Plex already has these shows, so then why would it be re-downloading them?

How can I verify that Riven is 100% connected to Plex and reading its DB?

---

Another idea: Could it be related to the fact that I have two Zurg's running? I have the Zurg on the Windows machine running, because it is where Plex is and I think I have to have that running beside each other because it is locally serving the files. And I have the Zurg running in DMB, because if I disable it there then Riven wont start. My thought initially was that this wouldn't matter, because Zurg simply is a mounter for RD, so if I am mounting the same filenames (and same Zurg config between them), it wouldn't matter? 

But now I am thinking, maybe this is the cause of the issue, because (I am speculating here) Riven is checking the filePATH of the files in the DB, and because Plex reports a Windows-oriented path, but Riven is connected to Zurg on a linux-oriented path, maybe the mismatch makes Riven think that the media doesn't actually exist in Plex?

Again I am speculating on the above. Perhaps it is the case that Riven doesn't check via the path, but only the fact that Plex reports the media exists (regardless of it's source). Could it be? But if that is the case, then I have no idea why Riven is re-downloading media.

---

Apologies this post is already long, so I will stop writing here. I am not sure what to do next, so I hope that someone could give me some indication on where to start diagnosing. 

My thought would be to move my Plex Server to a portainer stack beside DMB, but that is 1) very time consuming, which I don't have a lot of currently, 2) requires that I do a hardware passthrough of my GPU to Plex for transcoding, but I am already passing through in a separate VM, so that's its own headache, and 3) I'm unsure this would even solve the issue, after all the trouble and time. So I am asking for help here :)

**Comment by I-am-PUID-0** [comment]
Duplicates are a known issue with Riven for the moment. As an aside, Riven does not check the Plex library for existing content, like plex_debrid; it leverages its own database.

**Comment by profucius** [comment]
Wow. I was under the impression Riven checked Plex, why else does it need the Plex URL in the config?

Genuinely surprised it's not considered a high priority to them. But then perhaps not everyone uses DMM to even know how many duplicates they have.

I suppose there is no workaround? Is there any downside to just letting it create thousands of duplicates per week? Could that be a burden on Plex? (managing tens or even hundreds of copies of a single media, across hundreds of medias)
